compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -1551,7 +1550,7 @@ fn check_return_position_impl_trait_in_trait_bounds<'tcx>(
         for arg in fn_output.walk() {
             if let ty::GenericArgKind::Type(ty) = arg.unpack()
                 && let ty::Alias(ty::Opaque, proj) = ty.kind()
-                && tcx.def_kind(proj.def_id) == DefKind::ImplTraitPlaceholder
+                && tcx.is_impl_trait_in_trait(proj.def_id)
```

Comment:
I don't think we need this when the new lowering strategy is enabled -- can you verify that we're hitting `check_associated_type_bounds` for the synthetic associated types instead? Or if not, then can you find out why not? Or at least commit a `FIXME(-Z.....)` that explains that this fn should be redundant with the new lowering strategy.

Reply:
Ok, well it's clear why is not called. The calls to `check_associated_type_bounds` start in [check_well_formed](https://github.com/spastorino/rust/blob/9d89591622f16b8ef0a7e5c9d388fa039efa95cf/compiler/rustc_hir_analysis/src/check/wfcheck.rs#L124), which is called from [check_mod_type_wf](https://github.com/spastorino/rust/blob/9d89591622f16b8ef0a7e5c9d388fa039efa95cf/compiler/rustc_hir_analysis/src/check/wfcheck.rs#L1903) by iterating through the hir nodes, so of course this one is not there. The question would be how to nicely solve this.
I could do ...
```rust
for rpitit in items.definitions().filter(|id| tcx.opt_rpitit_info(id).is_some()) {
    tcx.ensure().check_well_formed(rpitit);
}
```

but doesn't sound very clean to me.

Reply:
> I could do ...

Sorry, but you didn't say where you would be adding that code?

I guess the problem here is that `hir_module_items()` doesn't include the synthetic associated type for each RPITIT. I'm not actually sure what the best approach to fixing this is -- maybe @cjgillot has some thoughts?

I don't think we can just modify the HIR item collector to also "visit" synethetic items... Or at least that would need some thinking about the best way of doing it. Maybe not though?

Anyways, this could at least use a FIXME explaining what the issue is currently, so we can revisit this before stabilization.

Reply:
In `check_mod_type_wf`